### ISSUE TITLE
type aware not working

### ISSUE THREAD

**Issue by Boshen** [oxc-project/zed-oxc#60]
> Not sure if it's related or not, but I can't get 0.4.1 to pick up lsp configuration directives at all. In the LSP Logs > Server Info it always reports `* Configuration: {}` now. Perhaps that's always been the case, but when the linter isn't reporting the same as when I run `oxlint --type-aware`. 
> 
> ```
> "lsp": {
>     "oxlint": {
>       "initialization_options": {
>         "options": {
>           "run": "onType",
>           "configPath": null,
>           "tsConfigPath": null,
>           "disableNestedConfig": false,
>           "unusedDisableDirectives": "warn",
>           "typeAware": true,
>         },
>       },
>     },
>   },
> ``` 

 _Originally posted by @coreyward in [#58](https://github.com/oxc-project/oxc-zed/issues/58#issuecomment-3700135928)_

**Comment by Boshen** [comment]
@nrayburn-tech I can confirm that type-aware is not working. I follow the instructions in README.md

**Comment by nrayburn-tech** [comment]
Seems to be working for me, reproduction attached.

<img width="1027" height="719" alt="Image" src="https://github.com/user-attachments/assets/bfc550fb-c9d7-43be-9053-0e1fa6d55981" />

[oxc-zed-60.zip](https://github.com/user-attachments/files/24389523/oxc-zed-60.zip)

**Comment by Boshen** [comment]
Ok this works. Shall we add an ideal setup in something like `examples/`.
